File: app/src/main/java/me/imzack/app/end/injector/component/AppComponent.java
package me.imzack.app.end.injector.component;

import android.content.Context;

import me.imzack.app.end.injector.module.AppModule;
import me.imzack.app.end.model.DataManager;

import org.greenrobot.eventbus.EventBus;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = AppModule.class)
public interface AppComponent {

    Context getContext();

    DataManager getDataManager();

    EventBus getEventBus();
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/AppComponent.kt
package net.zackzhang.app.end.injector.component

import android.content.Context
import dagger.Component
import net.zackzhang.app.end.injector.module.AppModule
import org.greenrobot.eventbus.EventBus
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {

    val context: Context

    val eventBus: EventBus
}
